src/Type/Definition/Type.php

```diff
@@ -55,105 +57,85 @@ abstract class Type implements JsonSerializable
     public $extensionASTNodes;
 
     /**
-     * @return IDType
-     *
      * @api
      */
-    public static function id()
-    {
-        return self::getStandardType(self::ID);
-    }
-
-    /**
-     * @param string $name
-     *
-     * @return (IDType|StringType|FloatType|IntType|BooleanType)[]|IDType|StringType|FloatType|IntType|BooleanType
-     */
-    private static function getStandardType($name = null)
+    public static function id() : ScalarType
```

Comment:
`IDType` should stay here, no?

```suggestion
    public static function id() : IDType
```

Reply:
Can't, unfortunately; users are allowed to replace any of the core built-int types with their own `CustomScalarType`, so without further refactoring of that system there's nothing more I can do at this point.